pkg/reconciler/common/read_releases.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	"golang.org/x/mod/semver"
+)
+
+const (
+	KoEnvKey = "KO_DATA_PATH"
+)
+
+// RetrieveManifestPath returns the manifest path for Knative based a provided version and component
+func RetrieveManifestPath(version, kcomponent string) string {
+	koDataDir := os.Getenv(KoEnvKey)
+	return filepath.Join(koDataDir, kcomponent, version)
+}
+
+func isFirstVersionMoreRecent(former, latter string) bool {
+	return semver.Compare(addMissingLetterV(former), addMissingLetterV(latter)) == 1
+}
+
+func addMissingLetterV(version string) string {
+	if version == "" || version[0] == 'v' {
+		return version
+	}
+	return fmt.Sprintf("v%s", version)
+}
+
+// ListReleases returns the all the available release versions available under kodata directory for Knative component.
+func ListReleases(kComponent string) []string {
+	releaseTags := []string{}
+	// List all the directories available under kodata
+	koDataDir := os.Getenv(KoEnvKey)
+	pathname := filepath.Join(koDataDir, kComponent)
+	fileList, err := ioutil.ReadDir(pathname)
+	if err != nil {
+		panic(err)
```

Comment:
Why panic here? Why not return errors for structured error handling? (Applies to all errors herein)

Reply:
@markusthoemmes I took the suggestions Jim has given.
It used to return errors with this function, but we do not expect any errors, since kodata structure is defined by us.